deployment/scripts/upgrade.sh

```diff
@@ -0,0 +1,452 @@
+    echo "$pod_name not stopped within $TIMES seconds"
+    return 1
+  else
+    return 0
+  fi
+}
+
+function waitForStart() {
+
+  local pod_name=$1
+
+  waitFor "[[ \"Running\" == \"\$(oc get pod $pod_name -o jsonpath='{.status.phase}')\" ]]"
+  if [[ $? -ne 0 ]]; then
+    echo "$pod_name not started within $TIMES seconds"
+    return 1
+  else
+    return 0
+  fi
+}
+
+# This lets us wait until the pod has been scheduled before we try to grab its name
+function getPodName() {
+
+  local dc=$1
+  local pod
+
+  pod=$(waitForValue "oc get pod -l deploymentconfig=$dc -o jsonpath='{.items[?(.metadata.labels.deploymentconfig==\"$dc\")].metadata.name}'")
+  if [[ $? -ne 0 ]]; then
+    return 1
+  else
+    echo "$pod"
+  fi
+}
+
+function checkESStarted() {
+
+  local pod=$1
+  local cluster_service
+
+  cluster_service=$(waitForValue "oc logs $pod | grep '\[cluster\.service[[:space:]]*\]'")
+  if [[ $? -ne 0 ]]; then
+    echo "Unable to find log message from cluster.service within $TIMES seconds"
+    return 1
+  fi
+
+  # If we detect a master, we won't recover indices
+  #  check for output from "[cluster.service " with "] detected_master ["
+# OR
+  # We're the master if we see this
+  #  check for output from "[cluster.service " with "] new_master ["
+  local master=$(echo $cluster_service | grep "new_master")
+  local non_master=$(echo $cluster_service | grep "detected_master")
+
+  # Check that we started
+  #  check for output from "[node " with "] started"
+
+  waitFor "[[ -n \"\$(oc logs $pod | grep '\[node[[:space:]]*\][[:space:]]*\[.*\][[:space:]]*started')\" ]]"
+  if [[ $? -ne 0 ]]; then
+    echo "Unable to find log message from node that ES started within $TIMES seconds"
+    return 1
+  fi
+
+  # Check that it recovered its indices after starting if we're a master
+  #  check for output from "[gateway" with "] recovered[:num:] indices into cluster state"
+  if [[ -n "$master" ]]; then
+    waitFor "[[ -n \"\$(oc logs $pod | grep '\[gateway[[:space:]]*\][[:space:]]*\[.*\][[:space:]]*recovered[[:space:]]*\[[[:digit:]]*\][[:space:]]*indices into cluster_state')\" ]]"
+    if [[ $? -ne 0 ]]; then
+      echo "Unable to find log message from gateway that ES recovered its indices within $TIMES seconds"
+      return 1
+    fi
+  else
+    # if we aren't master we should be started by now and should have detected a master
+    if [[ -z "$non_master" ]]; then
+      echo "Node isn't master and was unable to detect master"
+      return 1
+    fi
+  fi
+}
+
+function checkKibanaStarted() {
+
+  local pod=$1
+
+  waitFor "[[ -n \$(oc logs $pod -c kibana | grep 'Listening on 0.0.0.0:5601') ]]"
+  if [[ $? -ne 0 ]]; then
+    echo "Kibana was not able to start up within $TIMES seconds"
+    return 1
+  fi
+}
+
+function scaleDown() {
+
+  # check for a fluentd dc, if there isn't one it was either deleted or using a daemonset
+  local fluentd_dc=(`oc get dc -l logging-infra=fluentd -o jsonpath='{.items[*].metadata.name}'`)
```

Comment:
do you think there's really any probability there will be multiple fluentd DCs here?